src/nnet3/nnet-diagnostics.cc

```diff
@@ -121,24 +121,36 @@ void NnetComputeProb::ProcessOutputs(const NnetExample &eg,
         totals.tot_weight += tot_weight;
         totals.tot_objective += tot_objf;
       }
-      if (obj_type == kLinear && config_.compute_accuracy) {
+      // May not be meaningful if non-classification tasks
+      if (config_.compute_accuracy) {  
         BaseFloat tot_weight, tot_accuracy;
+        PerDimObjectiveInfo &totals = accuracy_info_[io.name];
```

Comment:
please rename this to acc_totals; now that the code is longer, it's starting to get confusing.